src/contracts/middleware/example/ECDSARegistry.sol

```diff
@@ -0,0 +1,260 @@
+    mapping(address => bool) public whitelisted;
+
+    // EVENTS
+    /**
+     * @notice Emitted upon the registration of a new operator for the middleware
+     * @param operator Address of the new operator
+     * @param socket The ip:port of the operator
+     */
+    event Registration(
+        address indexed operator,
+        string socket
+    );
+
+    /// @notice Emitted when the `whitelister` role is transferred.
+    event WhitelisterTransferred(address previousAddress, address newAddress);
+
+    /// @notice Modifier that restricts a function to only be callable by the `whitelister` role.
+    modifier onlyWhitelister {
+        require(whitelister == msg.sender, "BLSRegistry.onlyWhitelister: not whitelister");
+        _;
+    }
+
+    constructor(
+        IStrategyManager _strategyManager,
+        IServiceManager _serviceManager
+    )
+        RegistryBase(
+            _strategyManager,
+            _serviceManager,
+            1
+        )
+    {}
+
+    /// @notice Initialize whitelister and the quorum strategies + multipliers.
+    function initialize(
+        address _whitelister,
+        bool _whitelistEnabled,
+        uint256[] memory _quorumBips,
+        StrategyAndWeightingMultiplier[] memory _quorumStrategiesConsideredAndMultipliers
+    ) public virtual initializer {
+        _setWhitelister(_whitelister);
+        whitelistEnabled = _whitelistEnabled;
+
+        RegistryBase._initialize(
+            _quorumBips,
+            _quorumStrategiesConsideredAndMultipliers,
+            new StrategyAndWeightingMultiplier[](0)
+        );
+    }
+
+    /**
+     * @notice Called by the service manager owner to transfer the whitelister role to another address 
+     */
+    function setWhitelister(address _whitelister) external onlyServiceManagerOwner {
+        _setWhitelister(_whitelister);
+    }
+
+    /**
+     * @notice Callable only by the service manager owner, this function toggles the whitelist on or off
+     * @param _whitelistEnabled true if turning whitelist on, false otherwise
+     */
+    function setWhitelistStatus(bool _whitelistEnabled) external onlyServiceManagerOwner {
+        whitelistEnabled = _whitelistEnabled;
+    }
+
+    /**
+     * @notice Called by the whitelister, adds a list of operators to the whitelist
+     * @param operators the operators to add to the whitelist
+     */
+    function addToWhitelist(address[] calldata operators) external onlyWhitelister {
+        for (uint i = 0; i < operators.length; i++) {
+            whitelisted[operators[i]] = true;
+        }
+    }
+
+    /**
+     * @notice Called by the whitelister, removes a list of operators to the whitelist
+     * @param operators the operators to remove from the whitelist
+     */
+    function removeFromWhitelist(address[] calldata operators) external onlyWhitelister {
+        for (uint i = 0; i < operators.length; i++) {
+            whitelisted[operators[i]] = false;
+        }
+    }
```

Comment:
might be nice to abstract this away some of this copy-pasted stuff into a mix-in kind of contract

Reply:
What do you think the best way to do this is? A new WhitelistableRegistryBase? sorta hard to move the `onlyServiceManagerOwner`s into their own contract without doing so

Reply:
ah good point. that could make the inheritance tree (chain) pretty long then.
The mix-in could just define internal `_setWhitelister` and `_setWhitelistStatus` functions (we should probably have the latter, and add an event to it), and then the implementations could define external functions + attach auth to them? similar idea to what OZ does with their ERC20 design?

Reply:
Will separate in another PR